### DISCUSSION TITLE
Prevent freezing when appending text too much?

### DISCUSSION THREAD

**Discussion by AndnixSH** [augustoproiete/serilog-sinks-richtextbox#78] [Q&A]
WPF RichTextBox have a known issue that it freezes when appending multiple times in a short period of time. It can be replicated when looping like this

```
while (true)
{
    Log.Information("Hello, world!");
}
```

I can put a limit to show less infomation but that's not really a solution. I'm using 3rd party command line tools that prints a lot of outputs that RichTextBox can't handle
Does anyone know the best practices to prevent it?

**Comment by augustoproiete** [comment]
@AndnixSH I believe we'll need to move the implementation of this sink into a [PeriodicBatchingSink](https://github.com/serilog/serilog-sinks-periodicbatching) that throttles the amount of messages that are posted to the control (and waits for it to render before sending more).
